builtin/common/vector.lua

```diff
@@ -63,83 +79,96 @@ function vector.distance(a, b)
 end
 
 function vector.direction(pos1, pos2)
-	local x_raw = pos2.x - pos1.x
-	local y_raw = pos2.y - pos1.y
-	local z_raw = pos2.z - pos1.z
-	local x_abs = math.abs(x_raw)
-	local y_abs = math.abs(y_raw)
-	local z_abs = math.abs(z_raw)
-	if x_abs >= y_abs and
-	   x_abs >= z_abs then
-		y_raw = y_raw * (1 / x_abs)
-		z_raw = z_raw * (1 / x_abs)
-		x_raw = x_raw / x_abs
-	end
```

Comment:
I think the original bug was that this should have been an `elseif` ... `elseif` section